### ISSUE TITLE
Should prevent double click when adding new payment info

### ISSUE THREAD

**Issue by WeizhengSap** [SAP/cloud-commerce-spartacus-storefront#3102]
**Environment Details**
Spartacus: latest develop
Browser: Chrome 73.0.3683.86
OS: Mac Os
Happens on: http://dev-com-17.accdemo.b2c.ydev.hybris.com:4200/electronics-spa/en/USD/

**Steps to Reproduce**
Add items to cart
Proceed to checkout
During checkout, click on "Set as default" for payments
Go to MyAccount -> Payment Details -> There are 2 copies of that payment info

**Comment by WeizhengSap** [comment]
I tested this ticket on my local. The "create payment details" needs to send 2 requests, which may take several seconds to finish. So, I clicked the "continue" button, and I didn't see the response, then I clicked again. In the "My account", I can see 2 copies of the payment info. So, I think, we should disable the button until the requests complete to prevent the double click.

**Comment by qjimmy** [comment]
Note: Same issue with setting default address. If button is clicked twice, there will be 2 copies of the address info.

**Comment by qjimmy** [comment]
Fixed disabled button bug when error GlobalMessage appears

**Comment by WeizhengSap** [comment]
To make the behavior consistence with others, can we do this: if loading, display spinner.
```
<ng-template #loading>
  <div class="cx-spinner">
    <cx-spinner></cx-spinner>
  </div>
</ng-template>
```
So, instead of disable the button, spinner is displayed.

**Comment by WeizhengSap** [comment]
Same issue as this: https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/4280

**Comment by mazamcodes** [comment]
Fix: https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/5495

**Comment by bgambocjaviniar** [comment]
As mentioned in the PR, you should work from Jimmy's PR because the PR you have is just a subset of things that he has done. There's comments that Patrick left in it to tackle :)

**Comment by hackergil** [comment]
I can still reproduce this error. 

With normal network speed, you can see the spinner but there's a flicker. If you throttle the network, I can still do multiple requests.

**Comment by qjimmy** [comment]
Fixed the flicker issue #5678

**Comment by hackergil** [comment]
With the latest changes if I throttle the network to 3G I can click the button several times and I don't see the spinner anymore. The correct behavior would be

- The user clicks the button
- Spinner is shown (covering the button, or button is disabled)
- Once the payment method call returns successfully, we hide the spinner and show the order confirmation page
